### PR TITLE
feat: Service account snippet

### PR DESCRIPTION
**Pull request by BigBlackWolf**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: iam/cloud-client/snippets/list_service_accounts.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains code samples that demonstrate how to get list of service account.
+
+
+# [START iam_list_service_accounts]
+def list_service_accounts(project_id: str) -> dict:
```
Comment (irudykss) [inline iam/cloud-client/snippets/list_service_accounts.py:19]: also return/hint mismatch

File: iam/cloud-client/snippets/ima_create_key_test.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# PROJECT_ID = os.environ["IAM_PROJECT_ID"]
+
+
+# def test_retrieve_policy(
+#     capsys: "pytest.CaptureFixture[str]", deny_policy: str
+# ) -> None:
+#     # Test policy retrieval, given the policy id.
+#     iam_create_key(PROJECT_ID, deny_policy)
+#     out, _ = capsys.readouterr()
+#     assert re.search(f"Retrieved the deny policy: {deny_policy}", out)
```
Comment (irudykss) [inline iam/cloud-client/snippets/ima_create_key_test.py:30]: leftovers? :)

File: iam/cloud-client/snippets/delete_service_account.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains code samples that demonstrate how to get delete service account.
+
+# [START iam_delete_service_account]
+def delete_service_account(project_id: str, account: str) -> dict:
```
Comment (irudykss) [inline iam/cloud-client/snippets/delete_service_account.py:18]: also return/hint mismatch

File: iam/cloud-client/snippets/create_service_account.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+from google.cloud import iam_admin_v1
+from google.cloud.iam_admin_v1 import types
+
+
+def create_service_account(project_id: str, account_id: str, display_name: Optional[str] = None) -> None:
+    """
+    Creates a service account.
+    project_id: ID or number of the Google Cloud project you want to use.
+    account_id: ID or email which will be unique identifier of the service account
```
Comment (m-strzelczyk) [inline iam/cloud-client/snippets/create_service_account.py:28]: Service accounts come with a name of following format: `<username>@<project-name>.iam.gserviceaccount.com`

Is that parameter supposed to be the <username> part only, or the full e-mail format identifier? Or both are accepted?

File: iam/cloud-client/snippets/disable_service_account.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    request = types.DisableServiceAccountRequest()
+    name = f"projects/{project_id}/serviceAccounts/{account}"
+    request.name = name
+
+    iam_admin_client.disable_service_account(request=request)
+
+    request = types.GetServiceAccountRequest()
+    request.name = name
+
+    service_account = iam_admin_client.get_service_account(request=request)
```
Comment (m-strzelczyk) [inline iam/cloud-client/snippets/disable_service_account.py:39]: Let's return the service_account

File: iam/cloud-client/snippets/enable_service_account.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+# [START iam_enable_service_account]
+from google.cloud import iam_admin_v1
+from google.cloud.iam_admin_v1 import types
+
+
+def enable_service_account(project_id: str, account: str) -> None:
+    """
+    Enables a service account.
+    project_id: ID or number of the Google Cloud project you want to use.
+    account: ID or email which is unique identifier of the service account.
```
Comment (m-strzelczyk) [inline iam/cloud-client/snippets/enable_service_account.py:26]:
```suggestion
    Enables a service account.
    
    project_id: ID or number of the Google Cloud project you want to use.
    account: ID or email which is unique identifier of the service account.
```

File: iam/cloud-client/snippets/create_service_account.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    request.account_id = account_id
+    request.name = f"projects/{project_id}"
+
+    service_account = types.ServiceAccount()
+    service_account.display_name = display_name
+    request.service_account = service_account
+
+    account = iam_admin_client.create_service_account(request=request)
+
+    print(f"Created a service account: {account.email}")
```
[the remark this answers is not in the document]
Comment (BigBlackWolf) [reply iam/cloud-client/snippets/create_service_account.py:47]: Sure, added

**Comment by m-strzelczyk** [comment]
I guess the trick we need here is the good, old `time.sleep(5)` between disabling/enabling and doing a `get` on the account. Let the change propagate :)

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**